src/Qir/Runtime/lib/QIR/QirOutputHandling.cpp

```diff
@@ -0,0 +1,86 @@
+#include <cstdint>
+#include <sstream>
+#include "QirOutputHandling.hpp"
+#include "QirTypes.hpp"
+#include "QirRuntime.hpp"
+
+#define RECORD_TYPE "RESULT"
+#define RECORD_COLUMN_DELIMITER "\t"
+#define RECORD_PREFIX RECORD_TYPE RECORD_COLUMN_DELIMITER /* "RESULT" "\t" (== "RESULT\t") */
+
+
+static void PrintCStr(const char* cStr)
+{
+    QirString msg(cStr);
+    __quantum__rt__message(&msg);
+}
+
+extern "C"
+{
+    // Tuple Type Records
+
+    void __quantum__rt__tuple_start_record_output() // NOLINT
+    {
+        PrintCStr(RECORD_PREFIX "TUPLE_START");
+    }
+
+    void __quantum__rt__tuple_end_record_output() // NOLINT
+    {
+        PrintCStr(RECORD_PREFIX "TUPLE_END");
+    }
+
+
+    // Array Type Records
+
+    void __quantum__rt__array_start_record_output() // NOLINT
+    {
+        PrintCStr(RECORD_PREFIX "ARRAY_START");
+    }
+
+    void __quantum__rt__array_end_record_output() // NOLINT
+    {
+        PrintCStr(RECORD_PREFIX "ARRAY_END");
+    }
+
+
+    // Primitive Result Records
+
+    void __quantum__rt__result_record_output(Result res) // NOLINT
+    {
+        if (!res)
```

Comment:
To make this compatible with other result representations, this will need to use `__quantum__rt__result_get_zero` and `__quantum__rt__result_equal` rather than directly using it as a Boolean.